### DISCUSSION TITLE
problem install

### DISCUSSION THREAD

**Discussion by Queru** [hlorus/CAD_Sketcher#386] [Announcements]
hi all, 
i try to install the zip file of CAD_sketcher on Blender 3.5. 
but it doesn't work. 
have you got an idea ? 
<img width="652" alt="Capture d’écran 2023-05-23 à 14 15 09" src="https://github.com/hlorus/CAD_Sketcher/assets/7768345/bb86e12d-744c-46c3-8ce6-07299c9628a0">
thank you for your help
Pascal

**Comment by nwoodman** [comment]
I have the same issue, recent install.  Managed to get get the local blender version of python to update  
py-slvs via the python blender console using 

from pip._internal import main
main(['install','pandas'])

However I am still getting the following as per Queru above

Traceback (most recent call last):
  File "/Applications/Blender.app/Contents/Resources/3.5/scripts/modules/addon_utils.py", line 333, in enable
    mod = __import__(module_name)
  File "/Users/XXXX/Library/Application Support/Blender/3.5/scripts/addons/CAD_Sketcher-main/__init__.py", line 3, in <module>
    from . import icon_manager, global_data
  File "/Users/XXXX/Library/Application Support/Blender/3.5/scripts/addons/CAD_Sketcher-main/icon_manager.py", line 13, in <module>
    _icon_shader = Shaders.uniform_color_image_2d()
  File "/Users/XXXX/Library/Application Support/Blender/3.5/scripts/addons/CAD_Sketcher-main/shaders.py", line 93, in uniform_color_image_2d
    return GPUShader(
Exception: Shader Compile Error, see console for more details

**Comment by michaeloryl** [comment]
I am thinking this is a 3.5 specific problem because of the new GPU API.  I'm getting the same error message when trying to enable the add-on.

**Reply by nwoodman** [reply]
Cheers, I even triedvresetting to “default settings” to no avail.  Hopefully the devs are onto it.  Coming from Solidworks, I am keen to try out the parametric modelling aspects.  Blender certainly handles 3d scans nicely.

**Comment by AlejandroRivera** [comment]
I ran into this issue with my M1 Macbook Pro (2021). I worked around it by changing the GPU Backend (Settings > System > GPU Backend) from Metal to OpenGL.

**Reply by Queru** [reply]
i will try that. 
thank you

**Reply by nwoodman** [reply]
Thanks, changing the GPU backend worked for me

**Reply by psdcovers** [reply]
Same, would have not guessed switching to OpenGL would be a solution but it works.

**Reply by michaeloryl** [reply]
Worked for me, as well. Will likely have a (negative) performance impact on Blender as a whole, though.

**Reply by AlejandroRivera** [reply]
I was able to revert back to Metal after completing the setup of the addon, and everything remained working as expected (addon + blender)

**Comment by hlorus** [comment]
According to @AlejandroRivera this might just be a startup problem. As i don't have access to a mac it would be great if someone could test the newest version.

Here's the commit that should solve it:
https://github.com/hlorus/CAD_Sketcher/commit/0f2bb9ba559fb760b484c013032b02cf99d0d9df

**Comment by Spookycake** [comment]
I had the same problem, after installing the add-on, i went to: /home/.config/blender/3.6/scripts/addons/CAD_Sketcher-main/icons
and i just created the missing folder: icons. went back to blender enabled it and it worked. make sure you have the latest add-on.
![Screenshot from 2023-11-09 21-22-35](https://github.com/hlorus/CAD_Sketcher/assets/83324488/db97db65-151e-4f59-889f-4294c9ad3b31)

**Reply by Spookycake** [reply]
always read the error message carefully to better understand what the issue is.
